nemo_curator/modifiers/pii_modifier.py

```diff
@@ -16,7 +16,6 @@
 import pandas as pd
 
 from nemo_curator.modifiers import DocumentModifier
-from nemo_curator.pii.algorithm import PiiDeidentifier
```

Comment:
Also wanted to note that I want to avoid importing libraries just for the sake of type checking. Let's try to clean all those references up and either have them as a string or use the `from __future__ import annotations` to avoid this

Reply:
Agreed. Lets move to that world quickly. 

Hopefully with the upcoming test in https://github.com/NVIDIA/NeMo-Curator/pull/675 we dont have to worry about it a little less

Reply:
I still have to learn the nuances, but I think `from typing import TYPE_CHECKING` is supposed to solve some of that.